db/data/20211228070332_remove_sharp_and_octothorp_of_tag_name_head.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class RemoveSharpAndOctothorpOfTagNameHead < ActiveRecord::Migration[6.1]
+  BEGINNING_WITH_SHARP_OR_OCTOTHORP = '^(#|＃|♯)( |　)*.*'
+
+  def up
+    ActsAsTaggableOn::Tag.where('name ~* ?', BEGINNING_WITH_SHARP_OR_OCTOTHORP).find_each do |tag_to_remove_name_head|
+      head_removed_name = tag_to_remove_name_head.name.sub(/^(#|＃|♯)( |　)*/, '')
+      existing_tag_id = ActsAsTaggableOn::Tag.where(name: head_removed_name).pick(:id)
+      if existing_tag_id
+        ActsAsTaggableOn::Tag.transaction do
+          ActsAsTaggableOn::Tagging.transaction do
+            ActsAsTaggableOn::Tagging.where(tag_id: tag_to_remove_name_head.id).find_each do |tagging_to_be_replaced|
+              tagging_to_be_replaced.tag_id = existing_tag_id
+              tagging_to_be_replaced.save!(validate: false)
+            end
+            tag_to_remove_name_head.destroy
```

Comment:
transaction は「ブロック内で例外が起きた時にロールバックを実行」なので、 [destroy!](https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-destroy-21) を使っておいた方が良いです。

```suggestion
            tag_to_remove_name_head.destroy!
```

Reply:
@sinsoku 
なるほど、こうすれば失敗を検知できるようになるんですね！
（sinsoku さんが初めてご指摘をくださってから毎回、「なんでこんな幸せが舞い込んできているんだ…？」と思いながらご指摘を頂戴しています🙏）